server/bound_queries.h

```diff
@@ -143,7 +143,7 @@ struct BoundGroupByQuery {
 
     const Dataset & from;
     WhenExpression when;
-    std::shared_ptr<SqlExpression> where;
+    const SqlExpression & where;
```

Comment:
Fair point.  Probably that will need to be handled by documentation... although the other parameters are in the same boat.